src/main/java/tc/oc/pgm/controlpoint/ControlPointModule.java

```diff
@@ -2,81 +2,92 @@
 
 import com.google.common.collect.ImmutableList;
 import java.util.ArrayList;
+import java.util.Collection;
+import java.util.LinkedList;
 import java.util.List;
-import java.util.Set;
 import java.util.logging.Logger;
 import org.jdom2.Document;
 import org.jdom2.Element;
+import tc.oc.pgm.api.map.MapModule;
+import tc.oc.pgm.api.map.MapTag;
+import tc.oc.pgm.api.map.factory.MapFactory;
+import tc.oc.pgm.api.map.factory.MapModuleFactory;
 import tc.oc.pgm.api.match.Match;
+import tc.oc.pgm.api.match.MatchModule;
 import tc.oc.pgm.filters.FilterModule;
 import tc.oc.pgm.goals.GoalMatchModule;
-import tc.oc.pgm.goals.GoalModule;
-import tc.oc.pgm.map.MapModule;
-import tc.oc.pgm.map.MapModuleContext;
-import tc.oc.pgm.maptag.MapTag;
-import tc.oc.pgm.module.ModuleDescription;
 import tc.oc.pgm.regions.RegionModule;
 import tc.oc.pgm.teams.TeamModule;
 import tc.oc.pgm.util.XMLUtils;
 import tc.oc.xml.InvalidXMLException;
 
-@ModuleDescription(
-    name = "Control Points",
-    depends = {TeamModule.class, GoalModule.class, RegionModule.class, FilterModule.class})
-public class ControlPointModule extends MapModule<ControlPointMatchModule> {
-
-  private static final MapTag CONTROLPOINT_TAG = MapTag.forName("controlpoint");
+public class ControlPointModule implements MapModule<ControlPointMatchModule> {
 
+  private static final Collection<MapTag> TAGS =
+      ImmutableList.of(MapTag.create("controlpoint", "Control the Point", true, false));
```

Comment:
I feel that we should make the description a BaseComponent too. `Control The Point` should be translatable.

Reply:
In the future, not today.